components/list.js

```diff
@@ -1,82 +1,70 @@
 import React from "react";
-import DefineMap from "can-define/map/";
-import Component from "react-view-model/component";
-import Todo from "../models/todo";
+import { connect, ObserveObject, observer } from "ylem";
+import Todo from "../models/todo"
 
-import canBatch from "can-event/batch/batch";
-
-export const ViewModel = DefineMap.extend("TodoListVM", {
-	todos: Todo.List,
-	editing: Todo,
-	backupName: "string",
-	isEditing(todo) {
+class Store extends ObserveObject {
+    todos = new Todo.List([]);
+	editing;
+	backupName;
+	isEditing = todo => {
 		return todo === this.editing;
-	},
-});
-
-export default class List extends Component {
-	edit(todo) {
-		canBatch.start();
+    }
+    edit = todo => {
 		this.backupName = todo.name;
 		this.editing = todo;
-		canBatch.stop();
 	}
-
-	cancelEdit() {
+	cancelEdit = () => {
 		if (this.editing) {
 			this.editing.name = this.backupName;
 		}
-		this.editing = null;
+        this.editing = null;
 	}
-
-	updateName(e) {
+	updateName = e => {
 		e && e.preventDefault();
-
 		this.editing.save();
 		this.editing = null;
-	}
-
-	render() {
-		return (
-			<ul id="todo-list">
-				{ this.viewModel.todos && this.viewModel.todos.map((todo) => (
-					<li key={ todo.id } className={ [
-						"todo",
-						todo.complete && "completed",
-						this.viewModel.isEditing(todo) && "editing",
-					].filter(v => v).join(" ") }>
-						<div className="view">
-							<input
-								className="toggle"
-								type="checkbox"
-								checked={todo.complete}
-								onChange={ () => {
-									todo.complete = !todo.complete;
-									todo.save();
-								} }
-							/>
-							<label
-								onDoubleClick={ () => this.viewModel.edit(todo) }
-							>{todo.name}</label>
-							<button
-								className="destroy"
-								onClick={ () => todo.destroy() }
-							></button>
-						</div>
-						<form onSubmit={ (e) => this.viewModel.updateName(e) }>
-							<input
-								className="edit"
-								type="text"
-								value={todo.name}
-								onChange={ (e) => todo.name = e.target.value }
-								onBlur={ () => this.viewModel.cancelEdit() }
-							/>
-						</form>
-					</li>
-				)) }
-			</ul>
-		);
-	}
+    }
 }
 
-List.ViewModel = ViewModel;
+const List = observer(({todos, isEditing, edit, updateName, cancelEdit, editing, length}) => {
```

Comment:
does `observer()` need to be called here?  Can't `connect()` do this for us?

Reply:
Anyone else feel like this sort of destructuring doesn't make sense when you are dealing with instances of a Type.  We go through all these effort to `OO` things, just to have it torn apart.  I think it would be nice to know what is on the VM and what is not.

It seems most odd in this case, where there are a lot of different properties.  Do we expect folks to destructure every method and property above here?  Why have all that double work?

Reply:
I think `length` isn't used.

Reply:
`connect` is specifically for connecting a Store to a Component, whereas `observer` is for making a component observable-aware but not attaching a specific store (usa case: it will be passed a prop that is observable). They should be used in the same location in code (ie: at the export or the decorator) and should be mutually exclusive with eachother. Since you have the `connect` lower, this can just be a function (though the react way here would be to use a named function, not an arrow, for debugging purposes).

Reply:
As for destructuring, this is the react way. The react component should not care whether it is receiving a VM or plain props; it will grab the info it needs, and call the functions as needed. We don't *expect* that people will destructure, but most react devs will.

Reply:
Is this the React way? A casual glance at react’s tutorials don’t show this. They show this.props.prop. 

I think destructurig can make things confusing when there are lots of other variables. Knowing what is props and what is some variable is useful. You get that by seeing props.prop.

Reply:
https://reactjs.org/tutorial/tutorial.html shows this.props.key

https://reactjs.org/docs/components-and-props.html shows props.key

Reply:
In my experience both destructing and calling `this.props`  are commonly used. Any react developer will have seen both, and it will be more of a preference thing. Personally I like destructing because it allows you to explicitly state which props you're going to use